i3status 2.12 fails to build on MacOS

Hi

The latest release of i3status, 2.12, fails to build on MacOS (10.13.4). (Previous release builds fine.)

Is supporting MacOS something that you are interested in? (I hope it is!)

First thing that I notice is that on MacOS `socket.h` seems to be at `/usr/include/sys/socket.h`, and not `/sys/socket.h`. I can help with testing / debugging if you like?

```
$ make
 CC i3status.c
 CC src/auto_detect_format.c
 CC src/first_network_device.c
 CC src/general.c
 CC src/output.c
src/print_battery_info.c:113:11: warning: unused variable 'outwalk' [-Wunused-variable]
    char *outwalk = buffer;
          ^
1 warning generated.
 CC src/print_battery_info.c
src/print_cpu_temperature.c:54:12: warning: unused function 'read_temperature' [-Wunused-function]
static int read_temperature(char *thermal_zone, temperature_t *temperature) {
           ^
1 warning generated.
 CC src/print_cpu_temperature.c
src/print_cpu_usage.c:58:22: warning: unused variable 'curr_all' [-Wunused-variable]
    struct cpu_usage curr_all = {0, 0, 0, 0, 0};
                     ^
src/print_cpu_usage.c:59:20: warning: unused variable 'diff_total' [-Wunused-variable]
    int diff_idle, diff_total, diff_usage;
                   ^
src/print_cpu_usage.c:59:9: warning: unused variable 'diff_idle' [-Wunused-variable]
    int diff_idle, diff_total, diff_usage;
        ^
src/print_cpu_usage.c:45:25: warning: unused variable 'prev_all' [-Wunused-variable]
static struct cpu_usage prev_all = {0, 0, 0, 0, 0};
                        ^
4 warnings generated.
 CC src/print_cpu_usage.c
 CC src/print_ddate.c
 CC src/print_disk_info.c
 CC src/print_eth_info.c
 CC src/print_ip_addr.c
 CC src/print_ipv6_addr.c
 CC src/print_load.c
 CC src/print_path_exists.c
 CC src/print_run_watch.c
 CC src/print_time.c
 CC src/print_volume.c
src/print_wireless_info.c:489:57: error: use of undeclared identifier 'AF_INET'
    char *ipv4_address = sstrdup(get_ip_addr(interface, AF_INET));
                                                        ^
src/print_wireless_info.c:490:57: error: use of undeclared identifier 'AF_INET6'
    char *ipv6_address = sstrdup(get_ip_addr(interface, AF_INET6));
                                                        ^
src/print_wireless_info.c:612:5: warning: implicit declaration of function 'free' is invalid in C99
      [-Wimplicit-function-declaration]
    free(ipv4_address);
    ^
1 warning and 2 errors generated.
make: *** [src/print_wireless_info.o] Error 1
```

--- a/src/print_wireless_info.c
+++ b/src/print_wireless_info.c
@@ -16,6 +16,7 @@
 #endif
 
 #ifdef __APPLE__
+#include <sys/socket.h>
 #define IW_ESSID_MAX_SIZE 32
 #endif
 

